### PR TITLE
Implement GFPO

### PR DESCRIPTION
**Pull request by faresobeid**
https://www.arxiv.org/abs/2508.09726
Basically to bias the model to generate shorter reasoning traces, it basically only keeps the shortest N rollouts in a group. This way the other rollouts are not penalised, but the model is just naturally pushed to generate shorter traces

### CODE REVIEW

**Review by samsja — APPROVED** [review]

**Comment by mikasenghaas** [comment]
put to draft until we have ablations

File: src/prime_rl/orchestrator/advantage.py [hunk 1]
```diff
@@ -96,7 +98,21 @@ def compute_advantages(
     for group_rewards, group_lengths in zip(all_group_rewards, all_group_lengths):
         group_rewards_tensor = torch.tensor(group_rewards)
         group_lengths_tensor = torch.tensor(group_lengths)
-        group_advantages_tensor = compute_advantage(group_rewards_tensor, group_lengths_tensor)
+
+        if keep_n_shortest is not None:
+            k = min(max(keep_n_shortest, 0), group_rewards_tensor.numel())
```
Comment (mikasenghaas) [inline src/prime_rl/orchestrator/advantage.py:103]: could also fail loudly if the value is not in [0,G] imo, rather than doing these fallbacks but up to you

File: src/prime_rl/orchestrator/config.py [hunk 2]
```diff
@@ -453,6 +453,14 @@ class OrchestratorConfig(BaseSettings):
         ),
     ] = "drgrpo"
 
+    advantage_keep_n_shortest: Annotated[
```
Comment (mikasenghaas) [inline src/prime_rl/orchestrator/config.py:456]: wondering if we should break this out into a `AdvantageCnfig` which has a type and `keep_n_shortest`

**Review by mikasenghaas — COMMENTED** [review]
nice! some minor comments re configs. are we planning any ablations for this one?
